### PR TITLE
chore: cleanup expected keepers

### PR DESCRIPTION
**Pull request by skosito**
# Description

Cleaned up unused methods from expected keepers interfaces. Also executed `make mocks` to remove not needed mocks.

One small inconsistency I noticed is that cosmos-sdk modules keepers (for example bank, account keeper etc) are sometimes added as dependency through `NewAppModule`, and sometimes when module keeper is initialized with `NewXKeeper`, not sure if there is some particular reason for this, but i tried to clean that up too.

Closes: #1716 

## Type of change

Cleanup

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [x] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -474,10 +475,10 @@ func New(
 		groupmodule.NewAppModule(appCodec, app.GroupKeeper, app.AccountKeeper, app.BankKeeper, interfaceRegistry),
 		evm.NewAppModule(app.EvmKeeper, app.AccountKeeper, evmSs),
 		feemarket.NewAppModule(app.FeeMarketKeeper, feeSs),
-		crosschainmodule.NewAppModule(appCodec, app.ZetaCoreKeeper, app.StakingKeeper, app.AccountKeeper),
-		observermodule.NewAppModule(appCodec, *app.ZetaObserverKeeper, app.AccountKeeper, app.BankKeeper),
-		fungibleModule.NewAppModule(appCodec, app.FungibleKeeper, app.AccountKeeper, app.BankKeeper),
-		emissionsModule.NewAppModule(appCodec, app.EmissionsKeeper, app.AccountKeeper),
+		crosschainmodule.NewAppModule(appCodec, app.ZetaCoreKeeper),
```
Comment (kingpinXD) [inline app/app.go:478]: Not required, it might be a good idea to rename `app.ZetaCoreKeeper` to `app.CrosschainKeeper`

**Review by kingpinXD — APPROVED** [review]
lgtm , left a  minor comment

File: app/app.go (hunk 1 above)
Comment (lumtis) [reply app/app.go]: Yes, this is correct, we should remove all Zeta prefix and keep `<ModuleName>Keeper` for the convention. For the pointer, I think we can handle it in another PR

File: x/fungible/types/expected_keepers.go [hunk 2]
```diff
@@ -1,12 +1,11 @@
 package types
 
 import (
-	"context"
+	context "context"
```
Comment (lumtis) [inline x/fungible/types/expected_keepers.go:4]:
```suggestion
	"context"
```
I don't think it is necessary?

**Review by lumtis — APPROVED** [review]
Looks good! one minor comment

File: x/fungible/types/expected_keepers.go (hunk 2 above)
Comment (lumtis) [reply x/fungible/types/expected_keepers.go]: Goland? It is sometimes doing weird things to me as well sometime.

**Review by kingpinXD — APPROVED** [review]
you would need to add a line in the changelog.mg to get the changelog check to pass

**Pull request merged**